silver/models/documents/base.py

```diff
@@ -493,3 +521,72 @@ def total_before_tax_in_transaction_currency(self):
     def tax_value_in_transaction_currency(self):
         return sum([entry.tax_value_in_transaction_currency
                     for entry in self.entries])
+
+
+def create_transaction_for_document(document):
+    # get a usable, recurring payment_method for the customer
+    PaymentMethod = get_model('silver.PaymentMethod')
+    Transaction = get_model('silver.Transaction')
+
+    payment_methods = PaymentMethod.objects.filter(
+        canceled=False,
+        verified=True,
+        customer=document.customer
+    )
+    for payment_method in payment_methods:
+        try:
+            return Transaction.objects.create(document=document,
+                                              payment_method=payment_method)
+        except ValidationError:
+            continue
+
+
+@receiver(post_transition)
+def post_transition_callback(sender, instance, name, source, target, **kwargs):
+    if not isinstance(instance, BillingDocumentBase):
+        return
+
+    document = instance
+    setattr(document, '.recently_transitioned', target)
+
+    document.save()
+
+
+@receiver(post_save)
+def post_document_save(sender, instance, created=False, **kwargs):
+    if not isinstance(instance, BillingDocumentBase):
```

Comment:
`if not isinstance(instance, BillingDocumentBase) or not hasattr(instance, '.recently_transitioned'):`
This will improve readability a little bit.